src/Formik.tsx

```diff
@@ -384,20 +384,28 @@ export class Formik<Values = FormikValues> extends React.Component<
     value: any,
     shouldValidate: boolean = true
   ) => {
```

Comment:
Would be nice to have a return type defined, this would have helped detect the inconsistency issue I've mentioned below
```suggestion
  ) : Promise<void> => {
```